services/src/main/java/org/keycloak/vault/FilesPlainTextVaultProvider.java

```diff
@@ -11,46 +11,48 @@
 import java.nio.file.Path;
 import java.nio.file.StandardOpenOption;
 import java.util.EnumSet;
+import java.util.List;
 import java.util.Optional;
 
 /**
- * A text-based vault provider, which stores each secret in a separate file. The file name needs to match a
- * vault secret id (or a key for short). A typical vault directory layout looks like this:
+ * A text-based vault provider, which stores each secret in a separate file. The file name needs to match a vault secret id (or
+ * a key for short) and follows the format provided by the configured {@link VaultKeyResolver}. A typical vault directory
+ * layout looks like this:
  * <pre>
  *     ${VAULT}/realma__key1 (contains secret for key 1)
  *     ${VAULT}/realma__key2 (contains secret for key 2)
  *     etc...
  * </pre>
- * Note, that each key needs is prefixed by realm name. This kind of layout is used by Kubernetes by default
- * (when mounting a volume into the pod).
+ * Note, that in this case each key is prefixed by realm name. This particular kind of layout is used by Kubernetes by default
+ * (when mounting a volume into the pod) and can be used by selecting the {@code REALM_UNDERSCORE_KEY} resolver (which is
+ * the default resolver when none is defined). Other layouts are available through different resolvers.
  *
  * See https://kubernetes.io/docs/concepts/configuration/secret/
  * See https://github.com/keycloak/keycloak-community/blob/master/design/secure-credentials-store.md#plain-text-file-per-secret-kubernetes--openshift
  *
  * @author Sebastian Łaskawiec
  */
-public class FilesPlainTextVaultProvider implements VaultProvider {
+public class FilesPlainTextVaultProvider extends AbstractVaultProvider {
 
     private static final Logger logger = Logger.getLogger(MethodHandles.lookup().lookupClass());
 
     private final Path vaultPath;
-    private final String realmName;
 
     /**
      * Creates a new {@link FilesPlainTextVaultProvider}.
      *
      * @param path A path to a vault. Can not be null.
      * @param realmName A realm name. Can not be null.
      */
-    public FilesPlainTextVaultProvider(@Nonnull Path path, @Nonnull String realmName) {
+    public FilesPlainTextVaultProvider(@Nonnull Path path, @Nonnull String realmName, @Nonnull List<VaultKeyResolver> resolvers) {
+        super(realmName, resolvers);
         this.vaultPath = path;
-        this.realmName = realmName;
         logger.debugf("PlainTextVaultProvider will operate in %s directory", vaultPath.toAbsolutePath());
     }
 
     @Override
-    public VaultRawSecret obtainSecret(String vaultSecretId) {
-        Path secretPath = resolveSecretPath(vaultSecretId);
+    public VaultRawSecret obtainSecretInternal(String vaultSecretId) {
```

Comment:
```suggestion
    protected VaultRawSecret obtainSecretInternal(String vaultSecretId) {
```

Reply:
Good catch, changed this and also ElytronCSKeystoreProvider to protected - no need to expose this method as public